### PR TITLE
feat(consumer group): allow assigning plugins to consumer groups

### PR DESCRIPTION
**Pull request by programmer04**
**What this PR does / why we need it**:

It allows specifying plugins for consumer groups via annotation, a subset of plugins is supported this way by Gateway https://github.com/Kong/kong-ee/pull/5291. Adjust integration test - introduce proper testing scenario (it checks actual behavior now).

**Which issue this PR fixes**:

This is part of https://github.com/Kong/kubernetes-ingress-controller/issues/3728

**Special notes for your reviewer**:

- it imports `deck` directly from `main` branch to include unreleased yet changes introduced in https://github.com/Kong/deck/pull/959
- integration test `TestConsumerGroup` is always skipped for nonenterprise deployments and enterprise ones that use Kong Gateway `< 3.4` (unreleased yet). Hence `TestConsumerGroup` for this PR is skipped because a used image is `3.3` see:
           - [integration-tests / enterprise-postgres](https://github.com/Kong/kubernetes-ingress-controller/actions/runs/5806072841/job/15738261291?pr=4452#logs)
           - [integration-tests / enterprise-dbless](https://github.com/Kong/kubernetes-ingress-controller/actions/runs/5806072841/job/15738261461?pr=4452#logs)
  But you can check that integration test `TestConsumerGroup` passes (is not skipped) by inspecting the result of a run of on-demand run integration tests (triggered with the label `ci/run-nightly`). They uses latest Kong Gateway image - https://github.com/Kong/kubernetes-ingress-controller/pull/4452#issuecomment-1670809995:
           - [integration-tests-enterprise-postgres-nightly](https://github.com/Kong/kubernetes-ingress-controller/actions/runs/5806126819/job/15738402001#logs)
           - [integration-tests-enterprise-dbless-nightly](https://github.com/Kong/kubernetes-ingress-controller/actions/runs/5806126819/job/15738402245#logs)

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -76,7 +76,7 @@ Adding a new version? You'll need three changes:
 ### Added
 
 - **WIP** Introduce `KongConsumerGroup` CRD (supported by Kong Enterprise only)
-  [#4325](https://github.com/Kong/kubernetes-ingress-controller/pull/4325), [#4387](https://github.com/Kong/kubernetes-ingress-controller/pull/4387), [#4419](https://github.com/Kong/kubernetes-ingress-controller/pull/4419), [4437](https://github.com/Kong/kubernetes-ingress-controller/pull/4437)
+  [#4325](https://github.com/Kong/kubernetes-ingress-controller/pull/4325), [#4387](https://github.com/Kong/kubernetes-ingress-controller/pull/4387), [#4419](https://github.com/Kong/kubernetes-ingress-controller/pull/4419), [#4437](https://github.com/Kong/kubernetes-ingress-controller/pull/4437), [#4452](https://github.com/Kong/kubernetes-ingress-controller/pull/4452)
```
Comment (randmonkey) [inline CHANGELOG.md:79]: Please put one PR per line here.

File: test/integration/consumer_group_test.go [hunk 2]
```diff
@@ -31,87 +37,270 @@ func TestConsumerGroup(t *testing.T) {
...
+				"window_type": "fixed"
+			 }`,
+			rateLimitValue,
+		),
+	)
+	cleaner.Add(pluginRateLimit)
+
+	// Currently Gateway supports only one plugin per consumer group, read more
+	// https://konghq.atlassian.net/browse/FTI-5282, but it silently accepts many
+	// but it runs only one of them. So we have to be careful.
```
Comment (czeslavo) [inline test/integration/consumer_group_test.go:82]: I was thinking of adding a validation rule to the admission webhook to catch CGs with multiple plugins then, but if that's going to be fixed in a patch release of Kong 3.4, I think that's ok. I created an issue so we have this tracked somewhere.
```suggestion
	// Currently Gateway supports only one plugin per consumer group, read more
	// https://konghq.atlassian.net/browse/FTI-5282, but it silently accepts many
	// but it runs only one of them. So we have to be careful.
	// https://github.com/Kong/kubernetes-ingress-controller/issues/4472 tracks extending 
	// this test once the limitation on Kong side is fixed.
```

Comment (programmer04) [reply test/integration/consumer_group_test.go:82]: Perfect, thank you!

**Review by czeslavo — APPROVED** [review]

**Comment by randmonkey** [comment]
@programmer04 please re-run `make lint` to pass the linting check.

**Review by czeslavo — APPROVED** [review]

**Pull request merged**